experiments/tribler/channel_download.py

```diff
@@ -0,0 +1,400 @@
+from os import path as path
+from sys import path as pythonpath
+import time
+
+import posix
+
+from twisted.internet import reactor
+from twisted.internet.task import LoopingCall
+
+from gumby.experiments.TriblerDispersyClient import TriblerDispersyExperimentScriptClient, BASE_DIR
+from gumby.experiments.dispersyclient import main, DispersyExperimentScriptClient
+
+# TODO(emilon): Fix this crap
+pythonpath.append(path.abspath(path.join(path.dirname(__file__), '..', '..', '..', "./tribler")))
+pythonpath.append('.')
+
+from Tribler.Core.DownloadConfig import DefaultDownloadStartupConfig
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.community.channel.community import ChannelCommunity
+from Tribler.community.channel.preview import PreviewChannelCommunity
+from Tribler.Policies.credit_mining_util import TorrentManagerCM
+from Tribler.Core.RemoteTorrentHandler import RemoteTorrentHandler
+
+class ChannelDownloadClient(TriblerDispersyExperimentScriptClient):
+
+    def __init__(self, *argv, **kwargs):
+        super(ChannelDownloadClient, self).__init__(*argv, **kwargs)
+        from Tribler.community.allchannel.community import AllChannelCommunity
+        self.community_class = AllChannelCommunity
+
+        self._logger.setLevel(logging.DEBUG)
+
+        self.my_channel = None
+        self.joined_community = None
+
+        self.join_lc = None
+        self.dl_lc = {}
+
+        self.upload_dir_path = None
+
+        self.torrent_mgr = None
+        self.downloaded_torrent = {}
+
+        self.num_peers = -1
+
+        self.id_experiment = os.environ['EXPERIMENT_NAME'].replace(" ", "")
+
+    def start_session(self):
+        super(ChannelDownloadClient, self).start_session()
+        self.session_deferred.addCallback(self.__config_dispersy)
+        self.session_deferred.addErrback(self._logger.error)
+
+        self.num_peers = os.environ.get('das4_instances_to_run', -1)
+        if self.num_peers == -1:
+            self._logger.error("Cannot get var:das4_instances_to_run")
+            self.num_peers = os.environ.get('sync_subscribers_amount', -1)
+            if self.num_peers == -1:
+                self._logger.error("Cannot get var:sync_subscribers_amount")
+                self.num_peers = len(self.get_peers()) or 200
+
+    def __config_dispersy(self, session):
+
+        count_out = 0
+        while self._dispersy is None and count_out < 100:
+            time.sleep(1.0)
+            count_out += 1
+
+        self.session.lm.dispersy = self._dispersy
+        # self.session.lm.init()
+        self.session.get_dispersy = True
+
+        self._dispersy.define_auto_load(ChannelCommunity, self._my_member, (), {"tribler_session": self.session})
+        self._dispersy.define_auto_load(PreviewChannelCommunity, self._my_member, (), {"tribler_session": self.session})
+
+        from Tribler.Core.TFTP.handler import TftpHandler
+
+        self.session.lm.tftp_handler = TftpHandler(self.session, self.endpoint,
```

Comment:
Why create a custom TFTP handler here? This one is created when Tribler starts up.